app/src/main/java/net/a8pade8/passwordsaver/activities/MainActivity.kt

```diff
@@ -0,0 +1,52 @@
+package net.a8pade8.passwordsaver.activities
+
+import android.content.Intent
+import android.os.Bundle
+import android.view.View
+import android.widget.AdapterView
+import android.widget.AdapterView.OnItemClickListener
+import android.widget.ListView
+import androidx.appcompat.app.AppCompatActivity
+import net.a8pade8.passwordsaver.R
+import net.a8pade8.passwordsaver.data.LisViewAdapter
+import net.a8pade8.passwordsaver.data.Record
+import net.a8pade8.passwordsaver.data.getAllRecordsFromPasswords
+
+class MainActivity : AppCompatActivity() {
+    private var recordsListView: ListView? = null
+
+    @Suppress(names = ["UNCHECKED_CAST"])
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_main)
+        recordsListView = findViewById(R.id.listOfRes)
+        recordsListView!!.onItemClickListener = OnItemClickListener { adapterView: AdapterView<*>, view: View, position: Int, id: Long -> onItemClick(adapterView, view, position, id) }
+        listResourcesShow()
+    }
+
+    fun openAddRecordActivity(view: View?) {
+        val openAddRecordActivity = Intent(this, AddRecordActivity::class.java)
+        startActivity(openAddRecordActivity)
+    }
+
+    private fun listResourcesShow() {
+        val adapter = LisViewAdapter(this, getAllRecordsFromPasswords())
+        recordsListView!!.adapter = adapter
+    }
+
+    private fun openResourceView(id: Long) {
+        val openResourceViewActivity = Intent(this, ResourceViewActivity::class.java)
+        openResourceViewActivity.putExtra("id", id)
+        startActivity(openResourceViewActivity)
```

Comment:
Intent(this, ResourceViewActivity::class.java).let {
            it.putExtra("id", id)
            startActivity(it)
        }